File: dotnet-core/Tea/Actions/TextStrategy.cs
using CustomExtensions;
using OpenQA.Selenium;

namespace Tea
{
    public class TextStrategy : ActionStrategy
    {

        public TextStrategy(WebBrowser browser, string action) : base(browser, action) {}
        public override void Execute(string value)
        {
            Action act = null;

            By by = By.LinkText(value);
            switch (_action)
            {
                case SelectorOptions.Action.Click: act = new Click(by); break;
                case SelectorOptions.Action.TypeText: act = new TypeText(by); break;
                default: throw new System.Exception("Invalid action chosen!");
            }

            act.Perform(Browser, value);
        }
    }
}

File: java/tea-framework/src/main/java/com/tea/framework/Actions/TextStrategy.java
package com.tea.framework.Actions;

import com.tea.framework.SeleniumWrapper.Drivers.WebBrowser;
import org.openqa.selenium.By;

public class TextStrategy extends ActionStrategy {

    public TextStrategy(WebBrowser browser, String action) {
        super(browser, action);
    }

    @Override
    public void Execute(String value) {
        Action act = null;

        By by = By.linkText(value);
        switch (_action) {
            case SelectorOptions.Action.Click:
                act = new Click(by);
                break;
            case SelectorOptions.Action.TypeText:
                act = new TypeText(by);
                break;
            default:
                throw new IllegalArgumentException("Invalid action chosen!");
        }

        act.Perform(Browser, value);
    }
}
